graph/articulationpoints.go

```diff
@@ -0,0 +1,64 @@
+// Algorithm to identify articulation points in a connected graph
+// You can read more about articulation points here: https://en.wikipedia.org/wiki/Biconnected_component
+// and the implementation and explanation here: https://cptalks.quora.com/Cut-Vertex-Articulation-point
+
+package graph
+
+import "github.com/TheAlgorithms/Go/math/min"
+
+type apHelper struct {
+	is_ap              []bool
+	visited            []bool
+	child_cnt          []int
+	discovery_time     []int
+	earliest_discovery []int
+}
+
+func ArticulationPoint(graph Graph) []bool {
+	time := 0 // time variable to keep track of the time of discovery_time of a vertex
+
+	//initialize all the variables
+	apHelperInstance := &apHelper{
+		is_ap:              make([]bool, graph.vertices), // boolean array to mark articulation points
+		visited:            make([]bool, graph.vertices), // boolean array to mark visited vertices
+		child_cnt:          make([]int, graph.vertices),  // array to store the number of children of a vertex
+		discovery_time:     make([]int, graph.vertices),  // array to store the time of discovery of a vertex
+		earliest_discovery: make([]int, graph.vertices),  // array to store the earliest discovered vertex reachable from a vertex
```

Comment:
Some of these comments are unhelpful. Comments should describe the behaviour not the action. Anyone who can read code would know that for example `is_ap` is a slice (array - you call it) of bool. That is not really helpful. Instead if you want to comment, you should comment on why instead of what/how.